packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -179,6 +191,41 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
         }
     }
 
+    /**
+     * @notice Gets the latest proposal time for a perpetual contract's funding rate.
+     * @dev This method is designed to be helpful for proposers in projecting their rewards.
+     * @param perpetual perpetual contract whose proposal time the caller is querying.
+     * @return the proposal time of the current funding rate record or a pending proposal that has expired.
+     */
+    function getProposalTimeForContract(address perpetual) external view nonReentrantView() returns (uint256) {
+        FundingRateRecord storage fundingRateRecord = _getFundingRateRecord(perpetual);
+        if (_getProposalState(fundingRateRecord.proposal) == ProposalState.Expired) {
+            return fundingRateRecord.proposal.time;
+        } else {
+            return fundingRateRecord.proposeTime;
+        }
+    }
+
+    /**
+     * @notice Gets the reward rate per second for a perpetual contract.
+     * @dev This method is designed to be helpful for proposers in projecting their rewards. To get the reward
+     * rate for a specific funding rate proposal, you need to scale this method's return value, which is the
+     * perpetual contract's "base reward rate", by the time elapsed since the last proposal and the magnitude
+     * of change between the proposed and current funding rate.
+     * @param perpetual perpetual contract whose base reward rate the caller is querying.
+     * @return FixedPoint.Unsigned representing the base reward rate for the given contract. 0.01 would represent a
+     * reward rate of 1% per second.
+     */
+    function getRewardRateForContract(address perpetual)
+        external
+        view
+        nonReentrantView()
+        returns (FixedPoint.Unsigned memory)
```

Comment:
Just curious: what do you think the advantages/disadvantages would be for combining these two methods into one that returns the projected reward? (not suggesting that's what we should do -- just want to hear your thoughts)

Reply:
Interesting. That could be a good idea if we anticipate that `getRewardRate` and `getProposalTime` would only be used to calculate potential rewards. Do you think there'd be other reasons to call them?

Reply:
Yeah, not sure. I can't think of any other use case.

Reply:
For the sake of reducing interface size, I'll combine these for now into one method that returns projected rewards.